### ISSUE TITLE
3.0 - Manual order - taxes being added when I haven't set a customer address

### ISSUE THREAD

**Issue by ashleyfae** [easydigitaldownloads/easy-digital-downloads#8107]
### Expected behavior

When I manually create a new order and don't have a customer address selected, I expect the tax rate to be 0. It should only be greater than 0 if I select a customer address in a country that has a tax rate.

### Actual behavior

When I create an order and add a download before selecting an address, tax is being applied anyway.

### Steps to reproduce the behavior
1) Enable taxes. These are my tax settings:
![Screenshot from 2020-09-14 12-48-56](https://user-images.githubusercontent.com/6324272/93082345-ac01b680-f688-11ea-8a6b-82ee0b493085.png)
2) Go to Downloads > Orders > Add New.
3) Add a download to the order. You'll see this notice:
![Screenshot from 2020-09-14 12-46-02](https://user-images.githubusercontent.com/6324272/93082358-b58b1e80-f688-11ea-99f2-a8653b40c008.png)
The expectation here is that no tax will be set because you haven't set an address.
4) But after clicking `Add Download` you can see that tax has been applied:
![Screenshot from 2020-09-14 12-46-21](https://user-images.githubusercontent.com/6324272/93082417-d05d9300-f688-11ea-9c99-4e72f3107aa4.png)

### Information (if a specific version is affected):
PHP Version: 7.4

EDD Version (or branch): release/3.0

WordPress Version:

Any other relevant information:

**Comment by spencerfinnell** [comment]
@ashleyfae Do you know if you somehow still have a default tax rate setup?

It is still referenced in https://github.com/easydigitaldownloads/easy-digital-downloads/blob/release/3.0/includes/tax-functions.php#L162 though it was supposed to be removed in https://github.com/easydigitaldownloads/easy-digital-downloads/issues/7511

The value is cleared out through a fake setting (https://github.com/easydigitaldownloads/easy-digital-downloads/pull/7512/files#diff-0fc4cfad82ef0e4a14720904d33c326cR2617) but it looks like we may need to completely delete it or remove the reference in `edd_get_tax_rate()`

**Comment by ashleyfae** [comment]
@spencerfinnell This `(float) edd_get_option( 'tax_rate', 0 )` returns `0` for me. So I'm guessing that means it's not the issue?

**Comment by spencerfinnell** [comment]
@ashleyfae Thanks for confirming. I'll keep looking!

**Comment by spencerfinnell** [comment]
@ashleyfae Can you please share the the request headers when you select a download:

<img width="322" alt="Screen Shot 2020-09-15 at 1 20 16 PM" src="https://user-images.githubusercontent.com/491124/93243506-4976ec00-f756-11ea-90cd-3db3c528c3a8.png">

The calculation only uses `edd_calculate_tax()`:

https://github.com/easydigitaldownloads/easy-digital-downloads/blob/release/3.0/includes/ajax-functions.php#L1502

Do you have similar behavior if you checkout on the frontend with no address?

**Comment by ashleyfae** [comment]
Solved it! This was the issue for me: https://github.com/easydigitaldownloads/easy-digital-downloads/blob/release/3.0/includes/tax-functions.php#L184-L194

It was falling back to my `edd_get_shop_country()` setting, which was GB, and I have a GB tax rate set up.

This makes sense on the front-end, but I think our current admin wording makes it very unclear for manual orders. Given that the admin is free to choose whatever country they wish, I'm inclined to say we set `$fallback` to `false` to stop this behaviour. Also because that function uses the _current user's_ address as the first fallback, which is not what you want in a manual order (most of the time).
